apps/editing-toolkit/editing-toolkit-plugin/dotcom-fse/templates/class-rest-templates-controller.php

```diff
@@ -11,6 +11,28 @@
  * Based on `WP_REST_Blocks_Controller` from core
  */
 class REST_Templates_Controller extends \WP_REST_Posts_Controller {
+	/**
+	 * Override the get_items method.
+	 *
+	 * Gutenberg uses the route `/wp/v2/template_parts?wp_id=%s&context=edit` to retrieve the current post data when editing
+	 * a template or template part, so we handle that here.
+	 *
+	 * @see https://github.com/WordPress/gutenberg/blob/6c94fb7a233e849e0fff51deffba6ab2866c60f4/packages/edit-post/src/editor.js#L71
+	 *
+	 * @param WP_REST_Request $request API request for template parts.
+	 * @return WP_REST_Response
+	 */
+	public function get_items( $request ) {
```

Comment:
`NB:` It's curious to me that we need to override the get_**items** action to return a _singular_ `template_part` resource.

I see in [the provided link](https://github.com/WordPress/gutenberg/blob/6c94fb7a233e849e0fff51deffba6ab2866c60f4/packages/edit-post/src/editor.js#L71) in the inline comments that the post editor is bootstrapped via `getEntityRecords` instead of `getEntityRecord`. The changes were made in [this PR](https://github.com/WordPress/gutenberg/pull/27910/files#diff-d00cfe048041886ee9b16f6f42422317b6e770cbea20cbe0e4d0ca7c8e0ec0e1), which was a huge undertaking. I wonder if it was an oversight 🤔

Reply:
I don't have historical insight, but my guess is that it has to do with how the "id" of a template in the context of the rest API is `{stylesheet}//{template-name}`... where as "wp_id" is the ID of the template post in the database.

The `get_item` [method for the templates controller](https://github.com/WordPress/wordpress-develop/blob/ea49625d59f41e17312d4c08503cd2dd567f4bda/src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L192) uses the stylesheet based template id, so calling the `get_items` route with `wp_id` is the only way to request a template by post ID right now. A workaround of sorts... it might be worth proposing that the `get_item` handler function also accept a numerical post ID.